presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcConnectorFactory.java

```diff
@@ -58,7 +59,14 @@ public Connector create(String connectorId, Map<String, String> requiredConfig)
         requireNonNull(optionalConfig, "optionalConfig is null");
 
         try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
-            Bootstrap app = new Bootstrap(new JdbcModule(connectorId), module);
+            Bootstrap app = new Bootstrap(
+                    new JdbcModule(connectorId),
+                    module,
+                    installModuleIf(
+                            JdbcSecurityConfig.class,
+                            security -> "none".equalsIgnoreCase(security.getPostgreSqlSecuritySystem()),
```

Comment:
As we were talking about, the Postgres-specific security stuff shouldn't be here, but does it have to be Postgres-specific? The types of security (no access control, read-only, SQL) are defined at the SPI level, so that config can live in the JdbcSecurityConfig and not be database-specific. If you have to override exactly what SQL access control look like for each database, you can use Guice to bind different modules for postgres, mysql, etc.